test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeModule.cs

```diff
@@ -19,23 +22,20 @@ public class EdgeModule
         protected string deviceId;
         protected IotHub iotHub;
 
+        public string Id { get; }
+
         public EdgeModule(string id, string deviceId, IotHub iotHub)
         {
             this.deviceId = deviceId;
-            this.iotHub = iotHub;
             this.Id = id;
+            this.iotHub = iotHub;
         }
 
-        public string Id { get; }
-
-        public static Task WaitForStatusAsync(EdgeModule[] modules, EdgeModuleStatus desired, CancellationToken token)
+        public static Task WaitForStatusAsync(IEnumerable<EdgeModule> modules, EdgeModuleStatus desired, CancellationToken token)
         {
-            (string template, string[] args) FormatModulesList() => modules.Length == 1
-                ? ("module '{0}'", new[] { modules.First().Id })
-                : ("modules ({0})", modules.Select(module => module.Id).ToArray());
+            string[] moduleIds = modules.Select(m => m.Id).Distinct().ToArray();
```

Comment:
Do we care about device id in modules are not the same and even not same as device running this function?

Reply:
I don't think so. This is a low-level function that simply calls `iotedge list`, so it's assumed that you're passing modules for one device (since that's what `iotedge list` returns). The nice thing is that the new `EdgeDeployment` class creates instances of `EdgeModule` now, always for modules on a single device, so test writers can't make that mistake.